lib/screens/authenticate/login/sign_in_screen.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:brew_crew/screens/authenticate/login/sign_in_model.dart';
+import 'package:brew_crew/screens/authenticate/register/register_coordinator.dart';
+import 'package:brew_crew/screens/navigator/slide_right_route.dart';
+import 'package:brew_crew/shared/login_register_button.dart';
+import 'package:flutter/material.dart';
+import 'package:brew_crew/shared/constant.dart';
+import 'package:brew_crew/shared/loading_widget.dart';
+
+class SignInScreen extends StatelessWidget {
+  final SignInModel model;
+
+  SignInScreen({this.model});
+
+  @override
+  Widget build(BuildContext context) {
+    return model.loading
+        ? LoadingWidget()
+        : Scaffold(
+            appBar: _buildAppBar(context),
+            backgroundColor: Colors.brown[100],
+            body: _buildBody(),
+          );
+  }
+
+  Widget _buildAppBar(BuildContext context) {
+    return AppBar(
+      backgroundColor: Colors.brown[400],
+      title: Text('Sign in to Brew Crew'),
+      actions: [
+        FlatButton.icon(
+          icon: Icon(Icons.person),
+          label: Text('register'),
+          onPressed: () {
+            Navigator.pushReplacement(
+              context,
+              SlideRightRoute(page: RegisterCoordinator()),
+            );
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _buildBody() {
+    return Container(
+      padding: EdgeInsets.symmetric(vertical: 20.0, horizontal: 50.0),
+      child: Form(
+        key: model.formKey,
+        child: Column(
+          children: [
+            SizedBox(height: 20.0),
+            TextFormField(
+              controller: model.email,
+              decoration: textInputDecoration.copyWith(hintText: 'Email'),
+              validator: (val) => val.isEmpty ? 'Enter email' : null,
+            ),
+            SizedBox(height: 20.0),
+            TextFormField(
+              controller: model.password,
+              decoration: textInputDecoration.copyWith(hintText: 'Password'),
+              validator: (val) => val.length < 6
+                  ? 'Password must be minimum 6 chars long'
+                  : null,
+              obscureText: true,
+            ),
+            SizedBox(height: 20.0),
+            LoginRegisterButton(onTap: model.login, text: 'login'),
+            Text(
```

Comment:
Buttony powinny raczej być zunifikowane, tj powinien być obiekt Button a nie konkretny LoginRegisterButton

Reply:
jeśli masz kilka typów buttonów w aplikacji to np PrimaryButton, SecondaryButton

Reply:
mam nadzieje że w końcu dobrze XD
done